vxquery-rest/src/main/java/org/apache/vxquery/rest/VXQueryApplication.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.vxquery.rest;
+
+import org.apache.vxquery.rest.core.VXQuery;
+import org.apache.vxquery.rest.core.VXQueryConfig;
+import org.apache.vxquery.rest.exceptions.VXQueryRuntimeException;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Main class responsible for starting the {@link RestServer} and {@link VXQuery} classes.
+ *
+ * @author Erandi Ganepola
+ */
+public class VXQueryApplication {
+
+    private static final Logger LOGGER = Logger.getLogger(VXQueryApplication.class.getName());
+
+    private VXQuery vxQuery;
+    private RestServer restServer;
+
+    public VXQueryApplication() {
+        VXQueryConfig config = loadConfiguration();
+        vxQuery = new VXQuery(config);
+        restServer = new RestServer(vxQuery);
+    }
+
+    public synchronized void start() {
+        try {
+            LOGGER.log(Level.INFO, "Starting VXQuery");
+            vxQuery.start();
+            LOGGER.log(Level.INFO, "VXQuery started successfully");
+
+            LOGGER.log(Level.INFO, "Starting REST server");
+            restServer.start();
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Error occurred when starting application", e);
+            stop();
+            throw new VXQueryRuntimeException("Error occurred when starting application", e);
+        }
+    }
+
+    public synchronized void stop() {
+        try {
+            LOGGER.log(Level.INFO, "Stopping REST server");
+            restServer.stop();
+
+            LOGGER.log(Level.INFO, "Stopping VXQuery");
+            vxQuery.stop();
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Error occurred when stopping the application", e);
+        }
+    }
+
+    private VXQueryConfig loadConfiguration() {
```

Comment:
@prestoncarman Yes. I agree with you. As per my understanding, VXQuery server is started by `CCDriver` class. I think I need to write a new class wrapping `CCDriver`class. Is that so? 

Currently, starting cluster controllers, stopping them and starting node controllers are done by 3 different classes, `CCDriver`, `NCDriver` and `VXQueryClusterShutdown`. If I'm to start/stop REST server along with them, all of them needs to be taken to one place (At least cluster start/stop). Am I correct?